include/grpcpp/ext/otel_plugin.h

```diff
@@ -0,0 +1,99 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPCPP_EXT_OTEL_PLUGIN_H
+#define GRPCPP_EXT_OTEL_PLUGIN_H
+
+#include <grpc/support/port_platform.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <memory>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/strings/string_view.h"
+#include "opentelemetry/metrics/meter_provider.h"
+
+namespace grpc {
+
+namespace internal {
+class OpenTelemetryPluginBuilderImpl;
+}  // namespace internal
+
+namespace experimental {
+// Metrics
+absl::string_view OpenTelemetryClientAttemptStartedInstrumentName();
+absl::string_view OpenTelemetryClientAttemptDurationInstrumentName();
+absl::string_view
+OpenTelemetryClientAttemptSentTotalCompressedMessageSizeInstrumentName();
+absl::string_view
+OpenTelemetryClientAttemptRcvdTotalCompressedMessageSizeInstrumentName();
+absl::string_view OpenTelemetryServerCallStartedInstrumentName();
+absl::string_view OpenTelemetryServerCallDurationInstrumentName();
+absl::string_view
+OpenTelemetryServerCallSentTotalCompressedMessageSizeInstrumentName();
+absl::string_view
+OpenTelemetryServerCallRcvdTotalCompressedMessageSizeInstrumentName();
+
+class OpenTelemetryPluginBuilder {
+ public:
+  OpenTelemetryPluginBuilder();
+  // If `SetMeterProvider()` is not called, no metrics are collected.
```

Comment:
Comments that should appear in our doxygen doc need three slashes instead of two.

Reply:
ah, thanks!